cdci_data_analysis/analysis/job_manager.py

```diff
@@ -55,8 +55,9 @@ class Job(object):
     def __init__(self,
                  instrument_name,
                  work_dir,
-                 server_url,
-                 server_port,
+                 dispatcher_callback_url_base,
+                 dispatcher_host,
```

Comment:
A Job will be instantiated without using the 'bind' options. Those are only used when starting the flask application

Reply:
Indeed, but dispatcher_host is always equal to bind_host. The same for port.
The complexity here is only that callback is often some external url not known to dispatcher unless provided in the config. So that's the second place which specifies similar information.
But there is no need for a third place, dispatcher_host and dispatcher_port

Reply:
Then, is it correct to say that the 'dispatcher_callback_url_base' parameter is only used by the various plugins when they need to access certain functionalities of the dispatcher?

Reply:
>     * Then, is it correct to say that the 'dispatcher_callback_url_base' parameter is only used by the various plugins when they need to access certain functionalities of the dispatcher?

by various backends, not plugins. Plugins are part of the dispatcher. I understand now there was some confusion on this.

>     * Shall we then get rid of the dispatcher_host and dispatcher_port? And have the Job instantiated using the bind options?

yes. There are two similar configuration aspects:

* bind_host + bind_port, it is where dispatcher binds
* dispatcher_callback_url_base, it is where dispatcher can be found by the backends. It might  coincide with `http://{bind_host}:{bind_port}`, and this should be assumed if `dispatcher_callback_url_base == None`. But in general, and quite often, dispatcher_callback_url_base is different and known by other means.

Reply:
Ok thanks for the clarification, I will try to better explain in the configuration example.